### PR TITLE
Refs #32534 - dont use SmartProxyDynflowCore

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Remove usage of SmartProxyDynflowCore.

### CODE REVIEW

File: lib/smart_proxy_dynflow/action/shareable.rb [hunk 1]
```diff
@@ -12,14 +12,14 @@ def plan(input)
...
+        plan_action(Proxy::Dynflow::Callback::Action, callback, planned_action.output)
       end
     end
 
     private
 
     def on_proxy?
-      defined?(SmartProxyDynflowCore::Callback)
+      defined?(Proxy::Dynflow::Callback)
     end
```
Comment (adamruzicka) [inline lib/smart_proxy_dynflow/action/shareable.rb:23]: This should now always be true, the action cannot run anywhere else than on the smart proxy

Comment (ezr-ondrej) [reply lib/smart_proxy_dynflow/action/shareable.rb:23]: Do we want to remove that then and remove usage of it?
Or just return always `true` and have a followup?

Comment (ezr-ondrej) [reply lib/smart_proxy_dynflow/action/shareable.rb]: I feel the latter is what we want, so I went with it, I'll try to come up with follow-up

Comment (adamruzicka) [reply lib/smart_proxy_dynflow/action/shareable.rb]: It doesn't seem to be used outside of this repository so I'd be fine with removing it along with its usage in this pr

Comment (adamruzicka) [reply lib/smart_proxy_dynflow/action/shareable.rb]: Hah, github didn't refresh. Nevermind then, let's do it in a followup

**Review by adamruzicka — APPROVED** [review]
Works well, ACK

**Comment by adamruzicka** [comment]
Thank you @ezr-ondrej !

**Pull request merged**